File: 1911. Maximum Alternating Subsequence Sum.py
# 1911. Maximum Alternating Subsequence Sum


class Solution:
    def maxAlternatingSum(self, nums: List[int]) -> int:
        odd, even = 0, 0
        for i in range(len(nums)-1, -1, -1):
            curOdd = max(even + nums[i], odd)
            curEven = max(odd - nums[i], even)
            odd, even = curOdd, curEven
        
        return odd

File: 1911. Maximum Alternating Subsequence Sum.cpp
// 1911. Maximum Alternating Subsequence Sum

#include <vector>
#include <algorithm>

class Solution {
    public:
        long long maxAlternatingSum(std::vector<int>& nums) {
            long long odd = 0, even = 0;
    
            for (int n: nums) {
                even = std::max(even, odd + n);
                odd = std::max(odd, even - n);
            }
            return even;
        }
    };